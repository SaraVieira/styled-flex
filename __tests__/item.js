import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Item from '../src/item';

test('it works', () => {
  const tree = renderer.create(React.createElement(Item)).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('order', '0');
  expect(tree).toHaveStyleRule('flex-basis', 'auto');
  expect(tree).toHaveStyleRule('flex-grow', '0');
  expect(tree).toHaveStyleRule('flex-shrink', '1');
});

test('it works - order=1', () => {
  const tree = renderer
    .create(React.createElement(Item, { order: 1 }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('order', '1');
});

test('it works - order=-1', () => {
  const tree = renderer
    .create(React.createElement(Item, { order: -1 }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('order', '-1');
});

test('it works - basis=10', () => {
  const tree = renderer
    .create(React.createElement(Item, { basis: 10 }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-basis', '10');
});

test('it works - basis=-10', () => {
  const tree = renderer
    .create(React.createElement(Item, { basis: -10 }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-basis', '-10');
});

test('it works - grow=10', () => {
  const tree = renderer
    .create(React.createElement(Item, { grow: 10 }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-grow', '10');
});

test('it works - grow=-10', () => {
  const tree = renderer
    .create(React.createElement(Item, { grow: -10 }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-grow', '-10');
});

test('it works - shrink=10', () => {
  const tree = renderer
    .create(React.createElement(Item, { shrink: 10 }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-shrink', '10');
});

test('it works - noShrink', () => {
  const tree = renderer
    .create(React.createElement(Item, { noShrink: true }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-shrink', '0');
});

test('it works - shrink=-10', () => {
  const tree = renderer
    .create(React.createElement(Item, { shrink: -10 }))
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-shrink', '-10');
});
