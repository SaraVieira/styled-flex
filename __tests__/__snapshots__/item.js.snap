// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works - basis=-10 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-flex-basis: -10;
  -ms-flex-preferred-size: -10;
  flex-basis: -10;
}

<div
  className="c0"
/>
`;

exports[`it works - basis=10 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-flex-basis: 10;
  -ms-flex-preferred-size: 10;
  flex-basis: 10;
}

<div
  className="c0"
/>
`;

exports[`it works - grow=-10 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-box-flex: -10;
  -webkit-flex-grow: -10;
  -ms-flex-positive: -10;
  flex-grow: -10;
}

<div
  className="c0"
/>
`;

exports[`it works - grow=10 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-box-flex: 10;
  -webkit-flex-grow: 10;
  -ms-flex-positive: 10;
  flex-grow: 10;
}

<div
  className="c0"
/>
`;

exports[`it works - noShrink 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<div
  className="c0"
/>
`;

exports[`it works - order=-1 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
}

<div
  className="c0"
  order={-1}
/>
`;

exports[`it works - order=1 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

<div
  className="c0"
  order={1}
/>
`;

exports[`it works - shrink=-10 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-flex-shrink: -10;
  -ms-flex-negative: -10;
  flex-shrink: -10;
}

<div
  className="c0"
/>
`;

exports[`it works - shrink=10 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
  -webkit-flex-shrink: 10;
  -ms-flex-negative: 10;
  flex-shrink: 10;
}

<div
  className="c0"
/>
`;

exports[`it works 1`] = `
.c0 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  display: block;
}

<div
  className="c0"
/>
`;
